crates/matrix-sdk/src/event_cache/linked_chunk.rs

```diff
@@ -590,18 +590,17 @@ impl ChunkIdentifierGenerator {
     /// Generate the next unique identifier.
     ///
     /// Note that it can fail if there is no more unique identifier available.
-    /// In this case, `Result::Err` contains the previous unique identifier.
-    pub fn generate_next(&self) -> Result<ChunkIdentifier, ChunkIdentifier> {
+    /// In this case, this method will panic.
+    pub fn generate_next(&self) -> ChunkIdentifier {
         let previous = self.next.fetch_add(1, Ordering::Relaxed);
-        let current = self.next.load(Ordering::Relaxed);
 
         // Check for overflows.
         // unlikely — TODO: call `std::intrinsics::unlikely` once it's stable.
-        if current < previous {
-            return Err(ChunkIdentifier(previous));
+        if previous == u64::MAX {
+            panic!("No more chunk identifiers available. Congrats, you did it. 2^64 identifiers have been consumed.")
         }
 
-        Ok(ChunkIdentifier(current))
+        ChunkIdentifier(previous.saturating_add(1))
```

Comment:
No need for the saturating_add, since we checked it's not u64::MAX above:

```suggestion
        ChunkIdentifier(previous + 1)
```

Reply:
Right! Done.